src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -47,10 +47,13 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
         // TODO: this assumes single-dataset snapshots, will need to add a loop for multiple
         switch (snapshotReq.getContents().get(0).getMode()) {
             case BYASSET:
+                addStep(new CreateSnapshotValidateAssetStep(datasetService, snapshotService, snapshotReq));
```

Comment:
Could you add a comment mentioning why this reordering is necessary?

Reply:
We just want to validate before we create! The previous order was my error